domeneshop/datasource_domeneshop_domain.go

```diff
@@ -0,0 +1,57 @@
+package domeneshop
+
+import (
+	"fmt"
+
+	"github.com/MindTooth/go-domeneshop"
+	"github.com/antihax/optional"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func dataSourceDomeneshopDomain() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDomeneshopDomainRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
Should maybe be optional?

Reply:
Nope.  As we only accept an exact match.  Must validate result of array size > 1.